test/DualTest.jl

```diff
@@ -434,4 +434,10 @@ for N in (0,3), M in (0,4), T in (Int, Float32)
     end
 end
 
+let p = Partials((0.0,0.0)),
+    d = Dual(1.0, p)
+    @test Base.rtoldefault(typeof(d)) ≡ Base.rtoldefault(typeof(value(d)))
+    @test Dual(1.0-eps(Float64), p) ≈ d
+end
```

Comment:
Instead of testing this here, can you place it [here](https://github.com/tpapp/ForwardDiff.jl/blob/f90357d9d6a30789930be0832c109153b86d6d83/test/DualTest.jl#L135) instead, and test it using `FDNUM` and `NESTED_FDNUM` instances instead of making a new `Dual`? 

That way, it will be tested against different number types, number of partials, nested duals, etc. AFAICT, your implementation currently works in all those cases, so the tests should catch any regressions with those cases if somebody changes the code in the future.